src/urban_disco/grid/group.clj

```diff
@@ -0,0 +1,47 @@
+(ns urban-disco.grid.group
+  (:require [grid.constants :as constants]
+            [grid.neighbors :as neighbor]
+            [grid.grid      :as grid]
+            [grid.tile      :as tile])
+  (:gen-class))
+
+(defn find-different-group-neighbors [grid tile group]
+  (let [{:keys [top right bottom left]} (neighbor/neighbor-indexes tile)]
+    (reduce (fn [out neighbor-index]
+              (if (and (>= neighbor-index 0)
+                       (not (= group (:group (nth grid neighbor-index)))))
+                (conj out (nth grid neighbor-index))
+                out))
+            []
+            [top right bottom left])))
+
+(defn find-all-different-groups [grid tiles group]
+  (into [] (set 
+            (flatten 
+             (reduce (fn [out tile]
+                       (let [neighbors 
+                             (find-different-group-neighbors grid tile group)]
+                         (if (not-empty neighbors)
+                           (conj out neighbors)
+                           out)))
+                     []
+                     tiles)))))
+
+(defn remove-neighboring-wall [grid tile]
+    (let [neighbors (find-different-group-neighbors grid tile :goal)]
+        (if (not-empty neighbors)
+            (grid/remove-neighboring-walls grid tile (first neighbors))
+            grid)))
+
+(defn merge-groups [grid start goal]
+  (let [group-neighbors 
+        (find-all-different-groups grid start :start)
+        total-to-remove 
+        (-> group-neighbors count (* 80) (/ 100) Math/floor int)]
```

Comment:
I think I'm confused about the threshold, is this saying remove 80% of the walls between the start group and the end group? Is this the last step in the algorithm you linked?

Reply:
The last step of the algorithm was to remove as many walls between the start and end group as you wanted. In testing the generated grid, 80% consistently remove the amount of walls to end of with the type of maze that I was looking to build. Leaving more walls forced too much backtracking to go between the start and end groups.

I'm happy to change this if you know a better way of handling it.

Reply:
That makes sense I just wanted to make sure I was understanding it correctly.